### PR TITLE
[INLONG-6945][SDK] Unify metrics report model of SortSDK

### LINKED ISSUE #6945: [Feature][SDK] Unify metrics report model of SortSDK
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request

- Fixes #6945

### Motivation

Unify metrics report model of SortSDK to Inlong Common.

### Modifications

Support metrics report model of Inlong.
Remove the old metrics report model.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is already covered by existing tests

### Documentation

  - Does this pull request introduce a new feature?  yes

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java [hunk 1]
```diff
@@ -17,19 +17,27 @@
...
-    protected final StatManager statManager;
+    protected final String sortTaskId;
+
+    protected final SortSdkMetricItemSet metricItemSet;
 
-    public ClientContext(SortClientConfig config, MetricReporter reporter) {
+    public ClientContext(SortClientConfig config) {
         this.config = config;
-        this.statManager = new StatManager(this, reporter);
+        this.sortTaskId = config.getSortTaskId();
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java:38]: This PR has too many changes. Is it necessary to add some UTs to verify them?

Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java:38]: This PR includes modification of #6944. 
I will rebase it when #6944 is merged.

**Review by dockerzhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]
LGTM

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java [hunk 2]
```diff
@@ -38,29 +46,118 @@ public SortClientConfig getConfig() {
...
+    }
+
+    private SortSdkMetricItem getMetricItem(InLongTopic topic, int partitionId) {
+        Map<String, String> dimensions = new HashMap<>();
+        dimensions.put(SortSdkMetricItem.KEY_SORT_TASK_ID, sortTaskId);
+        if (topic != null) {
+            dimensions.put(SortSdkMetricItem.KEY_CLUSTER_ID, topic.getInLongCluster().getClusterId());
+            dimensions.put(SortSdkMetricItem.KEY_TOPIC_ID, topic.getTopic());
+        }
+        dimensions.put(SortSdkMetricItem.KEY_PARTITION_ID, String.valueOf(partitionId));
```
Comment (luchunliang) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java:151]: Add a configuration item for partitionId dimension.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java [hunk 3]
```diff
@@ -38,29 +46,118 @@ public SortClientConfig getConfig() {
...
+        SortSdkMetricItem metricItem = this.getMetricItem(null, -1);
+        metricItem.requestManagerParamErrorCount.incrementAndGet();
+    }
+
+    private SortSdkMetricItem getMetricItem(InLongTopic topic, int partitionId) {
+        Map<String, String> dimensions = new HashMap<>();
+        dimensions.put(SortSdkMetricItem.KEY_SORT_TASK_ID, sortTaskId);
+        if (topic != null) {
+            dimensions.put(SortSdkMetricItem.KEY_CLUSTER_ID, topic.getInLongCluster().getClusterId());
+            dimensions.put(SortSdkMetricItem.KEY_TOPIC_ID, topic.getTopic());
```
Comment (luchunliang) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java:149]: In pulsar cluster, dimensions has many topic name.
Add a configuration item for topic dimension.

Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java:149]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/ClientContext.java]: fixed, thx

**Review by luchunliang — APPROVED** [review]
LGTM

**Pull request merged**